sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -836,10 +836,6 @@ private async Task RunReceiveTaskAsync(CancellationToken cancellationToken)
                         {
                             await _messageHandlerSemaphore.WaitAsync(linkedHandlerTcs.Token).ConfigureAwait(false);
                             messageHandlerLockAcquired = true;
```

Comment:
This is unnecessary since it would never be true in the one place we are checking it - the catch block.